### PR TITLE
Sockets

### PR DESCRIPTION
**Pull request by Serwlashev**
Changed class structure, deleted Acceptor and Connector sockets, replaced platform-specific code to the PlatformUtils.h and WindowsUtils.cpp, LinuxUtils.cpp. Deleted code, which becomes unimportant after changing the socket's structure.

### CODE REVIEW

File: utils/include/WindowsUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include "stdafx.h"
+#include "PlatformUtils.h"
+
+#ifdef _WIN64
+
+bool WindowsLibraryInititializator::s_is_started_library = false;
```
Comment (AntonVoytov) [inline utils/include/WindowsUtils.cpp:6]: Rework this class according to our discussion

File: Sockets/BackEndMonitoringSockets/src/CSocketWrapper.cpp [hunk 2]
```diff
@@ -43,25 +41,26 @@ std::string CSocketWrapper::Receive(const int socket)
...
+bool CSocketWrapper::IsAllDataReceived(int msg_size,
+	int received_msg_size) const
 {
 	return msg_size == received_msg_size;
 }
 
-bool CSocketWrapper::SendMessageLength(const int socket,  size_t length)
+bool CSocketWrapper::SendMessageLength(const int socket,  int length)
 {
 	if (send(socket, reinterpret_cast<char*>(&length),
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/src/CSocketWrapper.cpp:65]: Check send and recv type, and use this return value to check if everything is ok

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
+	m_service_handler = InitServiceHandler();
 	m_pool = InitThreadPool(num_threads, m_event);
 }
 
 void CAcceptorWrapper::StartServer()
 {
 	std::cout << "Start server" << std::endl;
 	int handle = SOCKET_ERROR;
-	while (true)
+	while (!m_event.WaitFor(std::chrono::nanoseconds(1)))
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:21]: What is the point of this !m_event.WaitFor(std::chrono::nanoseconds(1) ??

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp [hunk 4]
```diff
@@ -1,36 +1,45 @@
...
-			WRITE_DEBUG_WITH_PARAMS(*m_logger, "Connected with a new socket ", 
-				handle);
-			m_pool->Enqueue([this, &handle]() {
-				m_service_handler->HandleEvent(handle, 
-					EventType::REQUEST_DATA);
-				});
-
+			m_pool->Enqueue([this, &handle]()
+			{
+				while (!m_event.WaitFor(std::chrono::nanoseconds(1)))
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:27]: The same

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp [hunk 5]
```diff
@@ -1,36 +1,45 @@
...
-	: m_event(event), m_logger(logger)
+	size_t num_threads, CEvent& event)
+	: m_event(event)
 {
 	m_server_acceptor = InitAcceptor(1111, "127.0.0.1");
-	m_stream = InitSocketWrapper(m_server_acceptor->GetHandle());
-	m_service_handler = InitServiceHandler(m_server_acceptor->GetHandle());
+	m_stream = InitSocketWrapper();
+	m_service_handler = InitServiceHandler();
 	m_pool = InitThreadPool(num_threads, m_event);
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:14]: CAcceptorWrapper should not handle pool thread

**Review by AntonVoytov — CHANGES_REQUESTED** [review]
Please fix the code according to our discussion at the meeting.
Please provide only the socket wrapper, socket, and base socket classes for the next review.

File: utils/include/WindowsUtils.cpp (hunk 1 above)
Comment (Serwlashev) [reply utils/include/WindowsUtils.cpp:6]: I made it yesterday, but it's on my PC. I deleted this class, added two initialize functions. Now I start the library in CAcceptorWrapper and CConnectorWrapper only once in constructor and finalize it in destructor.

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp (hunk 3 above)
Comment (Serwlashev) [reply Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:21]: It's a temporary solution to check event state, it'll be changed when Check() method will be added to CEvent

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp (hunk 4 above)
Comment (Serwlashev) [reply Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:27]: The same as previous

File: Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp (hunk 5 above)
Comment (Serwlashev) [reply Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:14]: fixed

**Comment by Crowy-o** [comment]
now, sokects moved to Dev. Pull request is useless now

**Pull request closed without merging**